### ISSUE TITLE
Does not work in scala 2.10

### ISSUE THREAD

**Issue by upio** [fommil/spray-json-shapeless#19]
I get compile errors in scala 2.10 but can successfully compile in scala 2.11, even when using the most simplistic example.

To reproduce the error:

```
// scalaVersion := "2.11.7" // compiles
scalaVersion := "2.10.6" // does not compile

libraryDependencies += "com.github.fommil" %% "spray-json-shapeless" % "1.2.0"
```

```
import spray.json._
import fommil.sjs.FamilyFormats._

package domain {
    sealed trait SimpleTrait
    case class Foo(s: String) extends SimpleTrait
    case class Bar() extends SimpleTrait
    case object Baz extends SimpleTrait
    case class Faz(o: Option[String]) extends SimpleTrait
}

import domain._

object Test {
    (Foo("foo"): SimpleTrait).toJson 
}

```

Compile Error:
```
[error] /Users/upio/personal/spray-json-shapeless-bug/src/main/scala/Test.scala:15: Cannot find JsonWriter or JsonFormat type class for domain.SimpleTrait
[error]     (Foo("foo"): SimpleTrait).toJson
```

**Comment by fommil** [comment]
You're accessing from a parent package. Please read the javadocs.

**Comment by upio** [comment]
I change it to a sibling package and I still can't compile.

```
import spray.json._
import fommil.sjs.FamilyFormats._

package domain {
    sealed trait SimpleTrait
    case class Foo(s: String) extends SimpleTrait
    case class Bar() extends SimpleTrait
    case object Baz extends SimpleTrait
    case class Faz(o: Option[String]) extends SimpleTrait
}

package use {
    import domain._
    
    object Test {
        (Foo("foo"): SimpleTrait).toJson 
    }    
}
```

Can you suggest how I would change this code to work in scala 2.10? It's almost word for word the example from the README.

**Comment by fommil** [comment]
You need to import shapeless too. I think. That's weird, I can't see anything obviously wrong. Can you perhaps start with the specs and try to break those?

**Comment by fommil** [comment]
try doing the longer form version of creating a `format`s package or object. It's more efficient to do that and assign it via `cachedImplicit` anyway.

**Comment by upio** [comment]
Can you re-open the issue? I'll see if I can break the specs and let you know.

What do you mean by a formats package and cachedImplicit?

**Comment by fommil** [comment]
this isn't a bug with spray-json-shapeless, it's just something in the compiler that you're going to have to workaround. It clearly does work for 2.10 because we're using it in ensime-server and the 2.10 tests pass. Link to examples is on the README.

**Comment by upio** [comment]
I figured it out by looking through this projects build definition. To fix the issue in scala-2.10 you need to add the paradise compiler plugin.

```
scalaVersion := "2.10.6"

libraryDependencies ++= Seq(
    "com.github.fommil" %% "spray-json-shapeless" % "1.2.0",
    compilerPlugin("org.scalamacros" % "paradise" % "2.0.1" cross CrossVersion.full)
)
```

Thanks for your help :)

Do you think it's worth mentioning this on the README?

**Comment by fommil** [comment]
Possibly, this is a transient dependency of shapeless and i guess compiler plugins are per project. Thanks for sharing! If you send a simple PR I'd be happy to merge. Just point to the shapeless docs, don't reproduce the instructions.
